test/src/androidTest/kotlin/io/realm/shared/EncryptionTests.kt

```diff
@@ -0,0 +1,121 @@
+ * or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.shared
+
+import io.realm.Realm
+import io.realm.RealmConfiguration
+import io.realm.util.PlatformUtils
+import test.Sample
+import kotlin.random.Random
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertFails
+
+/**
+ * This class contains all the Realm encryption integration tests that validate opening a Realm with an encryption key.
+ *
+ *  [RealmConfigurationTests] tests how the encryption key is passed to a [RealmConfiguration].
+ */
+class EncryptionTests {
+    private lateinit var tmpDir: String
+
+    @BeforeTest
+    fun setup() {
+        tmpDir = PlatformUtils.createTempDir()
+    }
+
+    @AfterTest
+    fun tearDown() {
+        PlatformUtils.deleteTempDir(tmpDir)
+    }
+
+    @Test
+    fun openEncryptedRealm() {
+        val key = Random.nextBytes(64)
+        val encryptedConf = RealmConfiguration
+            .Builder(
+                path = "$tmpDir/default.realm",
+                schema = setOf(Sample::class)
+            )
+            .encryptionKey(key)
+            .build()
+
+        // Initializes an encrypted Realm
+        Realm(encryptedConf).close()
+
+        // Should be possible to reopen an encrypted Realm
+        Realm(encryptedConf).close()
+    }
+
+    @Test
+    fun openEncryptedRealmWithWrongKey() {
+        val actualKey = Random.nextBytes(64)
+
+        // Initialize an encrypted Realm
+        val encryptedConf = RealmConfiguration
+            .Builder(
+                path = "$tmpDir/default.realm",
+                schema = setOf(Sample::class)
+            )
+            .encryptionKey(actualKey)
+            .build()
+        Realm(encryptedConf).close()
+
+        // Assert fails with no encryption key
+        assertFails {
+            val conf = RealmConfiguration.Builder(schema = setOf(Sample::class))
+                .path("$tmpDir/default.realm")
+                .build()
+            Realm(conf)
+        }
+
+        // Assert fails with wrong encryption key
+        val randomKey = Random.nextBytes(64)
+        assertFails {
+            val conf = RealmConfiguration.Builder(schema = setOf(Sample::class))
+                .path("$tmpDir/default.realm")
+                .encryptionKey(randomKey)
+                .build()
+
+            Realm(conf)
+        }
+    }
+
+    @Test
+    fun openUnencryptedRealmWithWrongKey() {
+        // Initialize an unencrypted Realm
+        val unencryptedConf = RealmConfiguration
+            .Builder(
+                path = "$tmpDir/default.realm",
+                schema = setOf(Sample::class)
+            )
+            .build()
+        Realm(unencryptedConf).close()
+
+        // Assert fails opening with encryption key
+        val randomKey = Random.nextBytes(64)
+        assertFails {
```

Comment:
We should probably use `assertFailsWith(XXX::class)` here to provide more context about exactly how it fails.